lesson32/src/js/archive-news.js

```diff
@@ -111,13 +110,31 @@ const createThumbnail = article => {
     return thumbnailWrapper;
 };
 
-const renderArticleList = data => {
+const renderArticleList = (data, category = "all") => {
     const newsList = createElementWithClassName("ul", "news__list");
+    
+    if(category !== "all") {
+        data = filterArticlesData(data, category); 
+    }
+
     const fragment = document.createDocumentFragment();
     data.forEach(item => {
         fragment.appendChild(createArticleCards(item));
     })
-    newsContent.appendChild(newsList).appendChild(fragment);
+    newsContent.replaceChildren(newsList);
+    newsList.appendChild(fragment);
+};
+
+const filterArticlesData = (data, category) => {
+    const filteredData = data.filter(item => item.category === category);
+    return filteredData;
+};
```

Comment:
IMO
I felt that naming it with the function name would make it more consistent.
```suggestion
const filterArticlesData = (articles, category) => {
    const filteredArticles = articles.filter(article => article.category === category);
    return filteredArticles;
};
```